staging/src/k8s.io/client-go/plugin/pkg/client/auth/openstack/openstack.go

```diff
@@ -145,11 +156,20 @@ func newOpenstackAuthProvider(clusterAddress string, config map[string]string, p
 		}
 	}
 
-	// TODO: read/persist client configuration(OS_XXX env vars) in config
+	getter := tokenGetter{}
+	var authOpt gophercloud.AuthOptions
+	configByte, err := json.Marshal(config)
+	if err == nil {
+		json.Unmarshal(configByte, &authOpt)
```

Comment:
no error handling here?

Reply:
needs a test, and an example in doc

Reply:
looks like some options are not serialized in the AuthOptions struct. IdentityEndpoint in particular seems to be ignored. How would the `Token()` method know what server to contact?

Reply:
> How would the Token() method know what server to contact?

L62 `func (t *tokenGetter) Token()` will use this `authOpt` to get an authenticated client, where `IdentityEndpoint` will be used.

Reply:
> looks like some options are not serialized in the AuthOptions struct. IdentityEndpoint in particular seems to be ignored.

Good catch. `AuthOptions` omits this filed. We need to retrieve this value explicitly.